homeassistant/components/shopping_list/config_flow.py

```diff
@@ -0,0 +1,32 @@
+"""Config flow to configure ShoppingList component."""
+from homeassistant import config_entries
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+
+class ShoppingListFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for ShoppingList component."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Init ShoppingListFlowHandler."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        self._abort_if_unique_id_configured()
```

Comment:
Is it working ?
The unique_id is not set here so it won't abort.

use : 
https://github.com/home-assistant/home-assistant/blob/c361358c6df5ddab515517b73ba74cee2ae01b19/homeassistant/components/icloud/config_flow.py#L86-L89
without the None check and set the unique_id as the DOMAIN.